File: front-end/src/web/domain/service/endereco.service.ts
import {Injectable} from '@angular/core';
import {EnderecoRepository} from '../repositories/endereco.repository';

@Injectable()
export class EnderecoService {

  constructor(private enderecoRepository: EnderecoRepository) {
  }

  public getAdressByCep(cep: string): Promise<any> {
    return this.enderecoRepository.getAdressByCep(cep);
  }

  public find(cidade: string, uf: string): Promise<any> {
    return this.enderecoRepository.find(cidade, uf);
  }
}

File: back-end/src/main/java/online/meavalia/domain/service/EnderecoService.java
package online.meavalia.domain.service;

import lombok.RequiredArgsConstructor;
import online.meavalia.domain.entity.endereco.Cidade;
import online.meavalia.domain.repository.EnderecoRepository;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class EnderecoService {

    private final EnderecoRepository enderecoRepository;

    public Optional<Cidade> find(final String cidade, final String uf) {
        return this.enderecoRepository.find(cidade, uf);
    }
}
